### PR TITLE
Minor eng cleanup

### PR DESCRIPTION
**Pull request by mikem8361**
Fix SourceLink validation failures.

Add back the PR merge build trigger

Add back setting DotNetPublishUsingPipelines

### CODE REVIEW

**Comment by mikem8361** [comment]
An example of a successful official build: 

https://dev.azure.com/dnceng/internal/_build/results?buildId=552988&view=results

File: azure-pipelines.yml [hunk 1]
```diff
@@ -10,18 +11,14 @@ pr:
...
     - THIRD-PARTY-NOTICES.TXT
-    - LICENSE.TXT
+    - License.txt
 
 variables:
   - name: _TeamName
     value: DotNetCore
-  - name: _DotNetArtifactsCategory
-    value: .NETCore
+  - name: _SignType
```
Comment (hoyosjs) [inline azure-pipelines.yml]: This should not be set in public.

Comment (mikem8361) [reply azure-pipelines.yml]: Why would setting it matter when _SignType is only used by the internal builds?

Comment (hoyosjs) [reply azure-pipelines.yml]: Not a strong stance. I just don't like setting stuff that could get accidentally used in a refactor.

**Pull request merged**